core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -549,6 +552,12 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId)
           | WorkerOutOfSpaceException e) {
         throw Throwables.propagate(e); // we shall never reach here
       }
+
+      // Check if block is pinned on commit
+      if (isPinned) {
+        updatePinnedInodes(Collections.singleton(BlockId.getFileId(blockId)));
```

Comment:
I think we need a different API, this is currently an override operation, not a union